src/lib/InstantSearch.ts

```diff
@@ -42,7 +55,25 @@ function defaultCreateURL() {
  * @fires Instantsearch#render This event is triggered each time a render is done
  */
 class InstantSearch extends EventEmitter {
-  constructor(options) {
+  public client: InstantSearchOptions['searchClient'];
+  public insightsClient: InsightsClient | null;
+  public indexName: string;
+  public helper: AlgoliaSearchHelper | null;
+  public mainHelper: AlgoliaSearchHelper | null;
+  public mainIndex: Index;
+  public started: boolean;
+  public templatesConfig: object;
+  public _stalledSearchDelay: number;
+  public _searchStalledTimer: any;
+  public _isSearchStalled: boolean;
+  public _searchParameters: PlainSearchParameters;
+  public _searchFunction?: InstantSearchOptions['searchFunction'];
+  public _createURL?: (params: SearchParameters) => string;
+  public _createAbsoluteURL?: (params: SearchParameters) => string;
```

Comment:
All these class variables don't need to be public. Perhaps we can make some private and add getters?

Reply:
you're right, but `_createAbsoluteURL` is actually being used by the index component, do you suggest making it properly public without the prefix?

Reply:
Note also that in this PR I didn't change any semantics of the existing JS, which is why I didn't make changes to the visibility